plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskList.php

```diff
@@ -168,10 +168,23 @@ public function is_hidden() {
 	 * @return bool
 	 */
 	public function is_visible() {
-		if ( ! $this->visible || $this->is_hidden() || ! count( $this->get_viewable_tasks() ) > 0 ) {
+		// If the task list is explicitly set to not be visible, return false.
+		if ( ! $this->visible ) {
 			return false;
 		}
-		return ! $this->is_hidden();
```

Comment:
The last `$this->is_hidden()` is redundant. Remove it to improve performance and make the code cleaner.